common/protob/messages-solana.proto

```diff
@@ -42,6 +42,23 @@ message SolanaAddress {
     required string address = 1; // Solana address as Base58 encoded string
 }
 
+/**
+ * @embed
+ */
+message SolanaTxTokenAccountInfo {
+    required string base_address = 1;
+    required string token_program = 2;
+    required string token_mint = 3;
+    required string token_account = 4;
```

Comment:
It could, but from the point of view of the client it's IMO easier if the client can set the data directly. The accounts in the transaction are built by the Solana SDK as instructions are added to the transaction object and the order depends also on the type of the address - signer, writeable, readonly. The logic in the code wouldn't change much IMO - just the protobuf object would be smaller, not sure how big of a deal that is.

Reply:
There's just one token address for the tuple of `(base_address (owner), token_program, mint, bump)` - solana takes the first bump (iterating down from 255) which takes the key pair off of the elliptic curve.